File: TaskVoteUI/src/app/model/task.ts
/*
interface Serializable<T> {
    deserialize( input: Object ): T;
}*/

export class Task  {    //implements Serializable<Task>{
    private id: string;
    private name: string;
    private upVote: number;
    private downVote: number;

    getId(): string {
        return this.id;
    }
    setId(id: string) : void {
       this.id = id; 
    }

    getName() : string {
        return this.name;
    }
    setName(name: string) : void {
        this.name = name;
    }

    getUpVote() : number {
        return this.upVote;
    }
    setUpVote(upVote: number) : void {
        this.upVote = upVote;
    }

    getDownVote() : number {
        return this.downVote;
    }
    setDownVote(downVote: number) : void {
        this.downVote = downVote;
    }

    /*
    deserialize( json: any ) {
        this.id = json.id;
        this.name = json.name;
        this.upVote = json.upVote;
        this.downVote = json.downVote;
        return this;
    }

    deserializeList(json: any) {
        let tasks: Array<Task> = [];

        if (json != null) {
          for (var i = 0; i < json.length; i++) {
            var data = json[i];
            var task: Task = new Task().deserialize(data);
            tasks.push(task);
          }
        }
        return tasks;
      }*/
}

File: TaskVoteApp/src/main/java/com/example/demo/model/Task.java
package com.example.demo.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
public class Task {
	
	@Id
	private String id;
	private String name;
	private Integer upVote;
	private Integer downVote;
	
	//Empty constructor is needed for Jakson to deserialize JSON
	public Task() {}
	
	@Override
	public String toString() {
		return "Task [id=" + id + ", name=" + name + ", upVote=" + upVote + ", downVote=" + downVote + "]";
	}

	public Task(String name, int upVote, int downVote) {
		this.name = name;
		this.upVote = upVote;
		this.downVote = downVote;
	}

	public String getId() {
		return id;
	}
	
	public String getName() {
		return name;
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	public Integer getUpVote() {
		return upVote;
	}
	
	public void setUpVote(Integer upVote) {
		this.upVote = upVote;
	}
	
	public Integer getDownVote() {
		return downVote;
	}
	
	public void setDownVote(Integer downVote) {
		this.downVote = downVote;
	}
}
